backend/src/apiserver/server/api_util.go

```diff
@@ -137,14 +160,57 @@ func GetResourceReferenceFromRunInterface(r interface{}) ([]*apiv1beta1.Resource
 	}
 }
 
-// Converts a yaml string into structpb.Struct (v2).
-func yamlStringToProtobufStruct(s string) (*structpb.Struct, error) {
-	st := &structpb.Struct{}
-	err := yaml.Unmarshal([]byte(s), st)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to convert a yaml string into a protobuf struct")
+// Converts pipeline spec in yaml string into structpb.Struct (v2).
+// The string may contain multiple yaml documents, where there has to be one doc for pipeline spec,
+// and there may be a second doc for platform specific spec.
+// If platform spec is empty, then return pipeline spec directly. Else, return a struct with
+// pipeline spec and platform spec as subfields.
+func yamlStringToPipelineSpecStruct(s string) (*structpb.Struct, error) {
+	if s == "" {
+		return nil, nil
+	}
+	var pipelineSpec structpb.Struct
+	var platformSpec structpb.Struct
+	hasPipelineSpec := false
+	hasPlatformSpec := false
```

Comment:
NIT: if we initialize `pipelineSpec` and `platformSpec` as pointers, we might be able to remove these booleans and check if `pipelineSpec == nil` or `platformSpec == nil`

Reply:
Similarly, see [reply](https://github.com/kubeflow/pipelines/pull/9024#discussion_r1145251806).